TripKitData/src/main/java/util/ReactiveSharedPreferences.kt

```diff
@@ -0,0 +1,17 @@
+package util
+
+import android.content.SharedPreferences
+import rx.Emitter
+import rx.Observable
+
+typealias PreferenceKey = String
+
+fun SharedPreferences.onChanged(): Observable<PreferenceKey> = Observable
+    .fromEmitter<PreferenceKey>({
+      val listener = SharedPreferences.OnSharedPreferenceChangeListener {
+        _, preferenceKey ->
+        it.onNext(preferenceKey)
+      }
+      registerOnSharedPreferenceChangeListener(listener)
+      it.setCancellation { unregisterOnSharedPreferenceChangeListener(listener) }
+    }, Emitter.BackpressureMode.BUFFER)
```

Comment:
There's a `ReactiveSharedPreferences` in `TripGoData` module already. Should we move it to `TripKitData` module instead of duplicating it here? If we move, can we keep the same package as it was which is `skedgo.tripgo.util`?

Reply:
Yeah. I moved that completely to TripKitData.